pkg/controller/change/network/controller.go

```diff
@@ -201,7 +204,7 @@ func (r *Reconciler) createDeviceChanges(networkChange *networkchange.NetworkCha
 	prevChange, err := r.networkChanges.GetByIndex(networkChange.Index - 1)
 	if err != nil && !errors.IsNotFound(err) {
 		return controller.Result{}, err
-	} else if err == nil && !hasDeviceChanges(prevChange) {
+	} else if prevChange != nil && !hasDeviceChanges(prevChange) {
 		return controller.Result{Requeue: controller.NewID(string(networkChange.ID))}, nil
```

Comment:
It does no harm to `Requeue` - but probably not needed, since we have a firehose of events coming in to the Watcher anyway